src/main/java/reactor/core/publisher/WorkQueueProcessor.java

```diff
@@ -771,8 +770,7 @@ public void run() {
 								readNextEvent(unbounded);
 							}
 							catch (Exception ce) {
-								if (isCancelled() || !isRunning() || !WaitStrategy
-										.isAlert(ce)) {
+								if (!isRunning() || !WaitStrategy.isAlert(ce)) {
```

Comment:
`!isRunning()` is more restrictive than simply `isCancelled()`, which is simply `!running.get()`, whereas `isRunning` checks both `running.get()` and a few other conditions. So the old code boils down to `!running.get() || !running.get() && !(...) || !WaitStrategy.isAlert(ce)`.

This can be immensely simplified to `!running.get() || !WaitStrategy.isAlert(ce)` though, right?